### PR TITLE
Tor/socks cleanups and hostname support.

### PR DESCRIPTION
**Pull request by rustyrussell**
How's this @Saibato ?  Works for me; basically after all the cleanups the actual patch to implement it is pretty short.

### CODE REVIEW

File: gossipd/gossip.c [hunk 1]
```diff
@@ -1939,10 +1939,27 @@ static struct io_plan *conn_init(struct io_conn *conn, struct reaching *reach)
...
+	switch (reach->addr.itype) {
+	case ADDR_INTERNAL_WIREADDR:
+		host = fmt_wireaddr_without_port(tmpctx,
+						 &reach->addr.u.wireaddr);
+		port = reach->addr.u.wireaddr.port;
+		break;
+	case ADDR_INTERNAL_SOCKNAME:
+	case ADDR_INTERNAL_ALLPROTO:
+	case ADDR_INTERNAL_AUTOTOR:
+		break;
```
Comment (ZmnSCPxj) [inline gossipd/gossip.c]: `abort();` instead?

**Comment by ZmnSCPxj** [comment]
ACK 8f39adaf6f2cf36e919eed299dab2c55d4f600b5

File: common/wireaddr.c [hunk 2]
```diff
@@ -299,13 +303,19 @@ bool wireaddr_from_hostname(struct wireaddr *addr, const char *hostname,
...
 
+	/* Tell them we wanted DNS and fail. */
+	if (no_dns) {
+		if (err_msg)
+			*err_msg = "Needed DNS, but lookups suppressed";
+		*no_dns = true;
+		return false;
+	}
+
 	memset(&hints, 0, sizeof(hints));
```
Comment (Saibato) [inline common/wireaddr.c]: Thanks! for doing this, confidence in that we care will give us later the feedback ln needs

File: gossipd/gossip.c (hunk 1 above)
Comment (rustyrussell) [reply gossipd/gossip.c]: The next line did that and more, by catching any invalid values and calling status_failed...

**Pull request merged**